install_config/router/default_haproxy_router.adoc

```diff
@@ -1487,6 +1487,94 @@ add routes across the namespaces.
 ====
 
 
+[[using-the-dynamic-configuration-manager]]
+== Using the Dynamic Configuration Manager
+
+The HAProxy router supports a dynamic configuration manager to bring certain
+types of routes online without requiring any haproxy reload downtime.
+The dynamic configuration manager support can be enabled by setting
+the `ROUTER_HAPROXY_CONFIG_MANAGER` environment variable to `true`.
+When the dynamic configuration manager is enabled, it will handle any route and
+endpoint lifecyle events such as route and endpoint `addition|deletion|update`.
+In the previous releases if any of these events caused the configuration to
+change, it resulted in the router requiring a full configuration re-write and a
+haproxy process reload.
+
+The dynamic configuration internally uses the haproxy socket and configuration
+API with a pool of pre-allocated routes and backend servers to dynamically
+achieve the same results without requiring a haproxy process reload.
+In the event the dynamic configuration manager is unable to configure haproxy
+dynamically, it forces a configuration re-write and a haproxy process reload.
+This can occur in cases like if a newly added route contains custom annotations
+that would cause a drift from the standard configuration (custom timeouts) or
+if the route requires custom TLS configuration.
+The pre-allocated pool of routes is created using route blueprints - the default
+set of blueprints supports unsecured routes, edge secured routes (without any
+custom TLS configuration) and passthrough routes.
```

Comment:
Can users `get` or `edit` the default set of blueprints? Does `ROUTER_BLUEPRINT_ROUTE_POOL_SIZE` also control the default blueprints? I mean if each route of default blueprints has 10 pools by default?

Reply:
Well it all depends on who `owns` the namespace for the blueprints and what access is allowed?
In the normal use case, the admin running the router would `own` that namespace and so an end user (one creating routes) would *_NOT_* have that access.

As re: the `ROUTER_BLUEPRINT_ROUTE_POOL_SIZE`, yes it controls the default blueprints as well as any custom ones the admin defines. 
And each one of those blueprints (default or custom)  ... for example blueprint-a  would have 10 (default) pre-defined backends - blueprint-a-1 through blueprint-a-10 - inside the haproxy config.